File: src/main/java/net/minheur/potoflux/utils/close/EventPostException.java
package net.minheur.potoflux.utils.close;

/**
 * Used to report an exception during event posting stage
 */
public class EventPostException extends RuntimeException {
    /**
     * Constructs a new EventPostException.
     * @param message exception message
     */
    public EventPostException(String message) {
        super(message);
    }

    /**
     * Constructs a new EventPostException.
     * @param message exception message
     * @param cause exception cause
     */
    public EventPostException(String message, Throwable cause) {
        super(message, cause);
    }

    /**
     * Constructs a new EventPostException.
     * @param cause exception cause
     */
    public EventPostException(Throwable cause) {
        super(cause);
    }

    /**
     * Constructs a new EventPostException.
     * @param message exception message
     * @param cause exception cause
     * @param writableStackTrace weather exception has writable stack trace
     */
    public EventPostException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }

    /**
     * Constructs a new EventPostException.
     */
    public EventPostException() {
        super();
    }
}

File: src/main/kotlin/net/minheur/potoflux/utils/close/EventPostException.kt
package net.minheur.potoflux.utils.close

/**
 * Used to report an exception during event posting stage
 */
class EventPostException : RuntimeException {
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
    constructor(cause: Throwable) : super(cause)
    constructor(message: String, cause: Throwable, enableSuppression: Boolean, writableStackTrace: Boolean) : super(message, cause, enableSuppression, writableStackTrace)
    constructor() : super()
}
